tests/Unit/PointwiseFunctions/Xcts/Test_CenterOfMass.cpp

```diff
@@ -99,55 +106,156 @@ void test_center_of_mass_surface_integral(const double distance) {
       // Get required fields on the interface
       const auto shifted_fields = solution.variables(
           shifted_coords,
-          tmpl::list<
-              Xcts::Tags::ConformalFactor<DataVector>,
-              Xcts::Tags::ConformalMetric<DataVector, 3, Frame::Inertial>,
-              Xcts::Tags::InverseConformalMetric<DataVector, 3,
-                                                 Frame::Inertial>>{});
+          tmpl::list<Xcts::Tags::ConformalFactor<DataVector>>{});
       const auto& conformal_factor =
           get<Xcts::Tags::ConformalFactor<DataVector>>(shifted_fields);
-      const auto& conformal_metric =
-          get<Xcts::Tags::ConformalMetric<DataVector, 3, Frame::Inertial>>(
-              shifted_fields);
-      const auto& inv_conformal_metric = get<
-          Xcts::Tags::InverseConformalMetric<DataVector, 3, Frame::Inertial>>(
-          shifted_fields);
-
-      // Compute outward-pointing unit normal.
-      const auto conformal_r = magnitude(inertial_coords, conformal_metric);
-      const auto conformal_unit_normal =
-          tenex::evaluate<ti::I>(inertial_coords(ti::I) / conformal_r());
 
       // Compute area element
-      const auto sqrt_det_conformal_metric =
-          Scalar<DataVector>(sqrt(get(determinant(conformal_metric))));
-      const auto conformal_area_element =
-          area_element(inv_jacobian, boundary_direction, inv_conformal_metric,
-                       sqrt_det_conformal_metric);
+      const auto flat_area_element =
+          euclidean_area_element(inv_jacobian, boundary_direction);
 
-      // Integrate
+      // Evaluate surface integral
       const auto surface_integrand = Xcts::center_of_mass_surface_integrand(
-          conformal_factor, conformal_unit_normal);
+          conformal_factor, inertial_coords);
       for (int I = 0; I < 3; I++) {
         surface_integral.get(I) += definite_integral(
-            surface_integrand.get(I) * get(conformal_area_element), face_mesh);
+            surface_integrand.get(I) * get(flat_area_element), face_mesh);
       }
     }
   }
 
   // Check result
-  auto custom_approx = Approx::custom().epsilon(10. / distance).scale(1.0);
+  auto custom_approx = Approx::custom().epsilon(TOLERANCE).scale(1.0);
   CHECK(get<0>(surface_integral) == custom_approx(0.));
   CHECK(get<1>(surface_integral) == custom_approx(0.));
   CHECK(get<2>(surface_integral) / mass == custom_approx(z_shift));
 }
 
+void test_infinite_volume_integral(const double distance,
+                                   const double z_shift) {
+  // Get Schwarzschild solution in isotropic coordinates.
+  const double mass = 1;
+  const Xcts::Solutions::Schwarzschild solution(
+      mass, Xcts::Solutions::SchwarzschildCoordinates::Isotropic);
+  const double horizon_radius = 0.5 * mass;
+
+  // Set up domain
+  const size_t h_refinement = 1;
+  const size_t p_refinement = 11;
```

Comment:
This is very high refinement. My understanding is that the accuracy of the integral is currently not limited by resolution, so can you lower this?

Reply:
This was the lowest resolution that I was able to find in order to keep the tolerance at $10^{-3}$. Figure 3.2.1 in these [convergence plots](https://iago-mendes.notion.site/Convergence-plots-90906819ac0742bcb8cdb8cb35292505) shows that the volume integral residual gets slightly better for higher resolution.

I think there must be some change that we can make in the calculation of the CoM integrals to get around these round-off errors, but this is probably something for a future PR.